analysis/common_variables.py

```diff
@@ -636,6 +640,349 @@ def generate_common_variables(index_date_variable, exposure_end_date_variable, o
+        #"tmp_cov_bin_history_rem_sleep_disorder_death", 
+    ),
+
+    # History of cognitive impairment
+    # Primary care - symptoms
+    cov_bin_history_cognitive_impairment=patients.with_these_clinical_events(
+        cognitive_impairment_symptoms_snomed,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+
+## History of MND
+    # Primary
+    tmp_cov_bin_history_motor_neurone_disease_snomed=patients.with_these_clinical_events(
+        motor_neurone_disease_snomed,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+    # HES
+    tmp_cov_bin_history_motor_neurone_disease_hes=patients.admitted_to_hospital(
+        with_these_diagnoses=motor_neurone_disease_icd10,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+    # ONS
+    # tmp_cov_bin_history_motor_neurone_disease_death=patients.with_these_codes_on_death_certificate(
+    #     motor_neurone_disease_icd10,
+    #     returning="binary_flag",
+    #     on_or_before=f"{index_date_variable} - 1 day",
+    #     return_expectations={"incidence": 0.1,
+    #     },
+    # ),
+
+    # Combined history of motor_neurone_disease
+    cov_bin_history_motor_neurone_disease = patients.minimum_of(
+        "tmp_cov_bin_history_motor_neurone_disease_snomed", "tmp_cov_bin_history_motor_neurone_disease_hes", 
+        #"tmp_cov_bin_history_motor_neurone_disease_death",
+    ),
+
+## History of migraine
+    # Primary
+    tmp_cov_bin_history_migraine_snomed=patients.with_these_clinical_events(
+        migraine_snomed,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+    # HES
+    tmp_cov_bin_history_migraine_hes=patients.admitted_to_hospital(
+        with_these_diagnoses=migraine_icd10,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+    # ONS
+    # tmp_cov_bin_history_migraine_death=patients.with_these_codes_on_death_certificate(
+    #     migraine_icd10,
+    #     returning="binary_flag",
+    #     on_or_before=f"{index_date_variable} - 1 day",
+    #     return_expectations={"incidence": 0.1,
+    #     },
+    # ),
+    # Combined history of migraine
+    cov_bin_history_migraine = patients.minimum_of(
+        "tmp_cov_bin_history_migraine_snomed", "tmp_cov_bin_history_migraine_hes", 
+        #"tmp_cov_bin_history_migraine_death",
+    ),
+
+    # History of Parkinson risk conditions (REM / Restless leg)
+    # Combined history of Parkinson risk conditions
+    cov_bin_history_parkison_risk_conditions = patients.minimum_of(
```

Comment:
As per meeting 09/01/2023, revise this based on above changes to `cov_bin_history_rem_sleep_disorder` and `cov_bin_history_restless_leg_syndrome`.

Reply:
Updated.